src/collectors/statsd/statsd.py

```diff
@@ -16,29 +16,45 @@
 milliseconds.
 """
 import re
-import sys
+import diamond.collector
 import diamond.metric
 import threading
 import socket
-import Queue
-from collections import namedtuple, defaultdict
+from collections import defaultdict
 
 
 ALIVE = True
 
-#remove non-alphanumeric characters
-def _clean_key(k):
+
+# remove non-alphanumeric characters
+def _clean_key(metric_name):
+    # type: (str) -> str
     return re.sub(
         r'[^a-zA-Z_\-0-9\.]',
         '',
         re.sub(
             r'\s+',
             '_',
-            k.replace('/', '-').replace(' ', '_')
+            metric_name.replace('/', '-').replace(' ', '_')
         )
     )
 
 
+# Spark statsd sink includes a unix timestamp in the metric name which causes high cardinality of metric names
+# by removing the timestamp we can significantly reduce this issue
+def _remove_spark_timestamp(metric_name):
+    # type: (str) -> str
+    return re.sub(r'^server.statsd.spark.local_\d+', 'server.statsd.spark', metric_name)
```

Comment:
Lets match local_ as a wildcard as well, so just so that we are being generic:
basically:

```
server\.statsd\.spark.[^\.]+\.
```

Reply:
also with regex pattern lets escape actual dots as dot is a wildcard char.